### ISSUE TITLE
Inference

### ISSUE THREAD

**Issue by jaideep11061982** [Elin24/SPDCN-CAC#1]
@Elin24 
thanks for this repo and congratulation for being 3rd in Object counting LB.
Could you let me know if you posted some where inference code
secondly, do the boxes are mandatory during training as well as inference ?

**Comment by Elin24** [comment]
Thanks for your attention.

A1: [Line-18](https://github.com/Elin24/SPDCN-CAC/blob/43602e13ebcb965af67a69504d0a11165c2e7bcc/run.sh#L18) of `run.sh` present how to eval SPDCN. 
In `main.py`, the default test set is the validation part of FSC-147. If you want to test it on the test part of FSC-147, the `'val'` should be replaced with `'test'` in [line 53](https://github.com/Elin24/SPDCN-CAC/blob/43602e13ebcb965af67a69504d0a11165c2e7bcc/main.py#L53).

A2: Boxes are required since they indicate what should be counted in given images (**exemplar-guided mode**). In the current case, other models (BMNet, CounTR, RCAC, etc.) also need box exemplars.

**Comment by jaideep11061982** [comment]
@Elin24  Thanks for the response.
How do we compute the offset for the Deform Conv2d  i.e what are we trying to pass on here
```
  def forward(self, x, bsize):
        b, c, h, w = x.shape
        g_offset = self.glob(bsize)
        g_offset = g_offset.view(b, -1, 1, 1).repeat(1, 1, h, w).contiguous()
        l_offset = self.local(x)
        offset = self.fuse(torch.cat((g_offset, l_offset), dim=1))

```

**Comment by Elin24** [comment]
`x` is the input feature map, `bsize` is (h, w) of exemplars. The structure of $\mathcal{C}$, $\mathcal{G}$ and $\mathcal{R}$ is  as following:
![image](https://user-images.githubusercontent.com/22393807/229019655-0e965ffc-3e1c-4c89-8528-f9c99bcefd46.png)
which is present in the [supplementary material](https://bmvc2022.mpi-inf.mpg.de/0313_supp.zip).
